examples/fungible/src/lib.rs

```diff
@@ -53,14 +53,75 @@ pub enum SessionCall {
 scalar!(AccountOwner);
 
 /// An account owner.
-#[derive(Clone, Copy, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
 pub enum AccountOwner {
     /// An account owned by a user.
     User(Owner),
     /// An account for an application.
     Application(ApplicationId),
 }
 
+impl Serialize for AccountOwner {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            AccountOwner::User(owner) => {
+                let key = format!("User:{}", owner);
+                serializer.serialize_str(&key)
+            }
+            AccountOwner::Application(app_id) => {
+                let key = format!("Application:{}", app_id);
+                serializer.serialize_str(&key)
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountOwner {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct AccountOwnerVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for AccountOwnerVisitor {
+            type Value = AccountOwner;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a string representing an AccountOwner")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                let parts: Vec<&str> = value.splitn(2, ':').collect();
+                if parts.len() != 2 {
+                    return Err(serde::de::Error::invalid_length(parts.len(), &self));
```

Comment:
Maybe `Error::custom` would be more appropriate. The actual error here is just that the string didn't contain a colon.